src/snowflake/snowpark/table.py

```diff
@@ -655,3 +661,11 @@ def merge(
             if block
             else result
         )
+
+    def drop_table(self):
+        """Drops the table from the Snowflake database.
+
+        Note that subsequent API such as ``select()``, ``collect()`` on this ``Table`` instance and the derived DataFrame will raise errors because the underlying
+        table in the Snowflake database no longer exists.
+        """
+        self._session.sql(f"drop table {self.table_name}").collect()
```

Comment:
I suggest to use `self._session._run_query`, which are usually used to run some internal queries in our codebase.

Reply:
I remember `_run_query` still throws a ProgrammingError?
We better change this first then update here. Now `session.sql` raises the correct error.

Reply:
Yea you're right. And here you might want to use internal_collect() (and for all other internal session.sql...)

Reply:
I changed to `_internal_collect_with_tag_no_telemetry`